File: src/main/resources/static/currency/currencyService.js
var services = angular.module('currency.services', [ 'ngResource' ]);

services.service('currencyService', [ '$http', function($http) {

	this.findAllCountries = function(){
		return $http.get("/country");
	}	
	
	this.findAll = function() {
		return $http.get("/currency");
	}
	
	this.save = function(currency) {
		return $http.post("/currency",currency);
	}
	
	this.next = function(id) {
		return $http.get("currency/nextCurrencies/" + id);
	}
	
	this.delete = function(currency){
		return $http.get('/currency/deleteCurrency/' + currency)
	}
	this.search = function(entity){
		return $http.post('/currency/search',entity);
	}
	this.update = function(currency){
		return $http.put("/currency/update", currency);
	}

	
}]);

File: src/main/java/bank/currency/CurrencyService.java
package bank.currency;

import java.util.List;

public interface CurrencyService {
	public List<Currency> findAll();

	public Currency save(Currency currency);

	public Currency findOne(Long id);
	
	public void delete(Long id);
	
	public List<Currency> search(Currency currency);
	
	public Currency findByCode(String code);
}
